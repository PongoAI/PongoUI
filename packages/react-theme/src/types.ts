import type { ColorSet } from '@pongo-ui/react-theme-generator';

export type Theme = ColorTokens & Fonts;

/**
 * Color theme tokens.
 */
export type ColorTokens = { textColor: string; canvasColor: string } & ColorSet<
  'inherit' | 'brand' | 'secondary' | 'success' | 'danger' | 'social' | 'warning' | 'info'
>;

export type Fonts = FontFamilyTokens & FontOpacityTokens & FontSizeTokens & FontLineHeightTokens & FontWeightTokens;

/**
 * Font family tokens.
 */
export type FontFamilyTokens = {
  baseFont: string;
  monospaceFont: string;
};

/**
 * Font opacity tokens
 */
export type FontOpacityTokens = {
  display: string;
  subHeadline: string;
  body: string;
  caption: string;
  title: string;
  header1: string;
  header2: string;
  header3: string;
};

/**
 * Font size tokens.
 */
export type FontSizeTokens = {
  fontSize100: string;
  fontSize200: string;
  fontSize300: string;
  fontSize400: string;
  fontSize500: string;
  fontSize600: string;
  fontSize700: string;
  fontSize800: string;
  fontSize900: string;
  fontSize1000: string;
};

/**
 * Font line height tokens.
 */
export type FontLineHeightTokens = {
  lineHeight100: string;
  lineHeight200: string;
  lineHeight300: string;
  lineHeight400: string;
  lineHeight500: string;
  lineHeight600: string;
  lineHeight700: string;
  lineHeight800: string;
  lineHeight900: string;
  lineHeight1000: string;
};

/**
 * Font weight tokens.
 */
export type FontWeightTokens = {
  thin: number;
  light: number;
  regular: number;
  medium: number;
  bold: number;
  black: number;
};
