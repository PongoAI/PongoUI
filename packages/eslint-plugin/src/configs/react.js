/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
