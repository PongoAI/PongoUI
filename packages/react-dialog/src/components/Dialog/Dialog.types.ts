import * as React from 'react';
import type { PopperVirtualElement, usePopperMouseTarget } from '@fluentui/react-positioning';
import type { PortalProps } from '@fluentui/react-portal';

/**
 * Events handled by the Dialog.
 */
export type OpenDialogEvents =
  | MouseEvent
  | TouchEvent
  | React.MouseEvent<HTMLElement>
  | React.KeyboardEvent<HTMLElement>
  | React.FocusEvent<HTMLElement>;

/**
 * Data attached to open/close events
 */
export type OnOpenChangeData = { open: boolean };

type DialogCommons = Pick<PortalProps, 'mountNode'> & {
  /**
   * Controlled value to handle the opening of the Dialog.
   */
  open: boolean;

  /**
   * Uncontrolled value to handle the opening of the Dialog.
   */
  defaultOpen?: boolean;

  /**
   * Call back when the component requests to change value
   * The `open` value is used as a hint when directly controlling the component
   */
  onOpenChange?: (e: OpenDialogEvents, data: OnOpenChangeData) => void;

  /**
   * Should trap focus
   */
  trapFocus?: boolean;
};

export type DialogProps = Partial<DialogCommons> & {
  children: [JSX.Element, JSX.Element] | JSX.Element;
};

export type DialogState = DialogCommons &
  Pick<DialogProps, 'children'> & {
    /**
     * Callback to open/close the Dialog
     */
    setOpen: (e: OpenDialogEvents, open: boolean) => void;

    /**
     * Callback to toggle the open state of the Dialog
     */
    toggleOpen: (e: OpenDialogEvents) => void;

    /**
     * Ref of the DialogTrigger
     */
    triggerRef: React.MutableRefObject<HTMLElement | null>;

    /**
     * Ref of the DialogSurface
     */
    contentRef: React.MutableRefObject<HTMLElement | null>;

    /**
     * Anchors the popper to the mouse click for context events
     */
    contextTarget: PopperVirtualElement | undefined;

    /**
     * A callback to set the target of the popper to the mouse click for context events
     */
    setContextTarget: ReturnType<typeof usePopperMouseTarget>[1];

    dialogTrigger?: React.ReactElement;

    dialogSurface?: React.ReactElement;
  };
