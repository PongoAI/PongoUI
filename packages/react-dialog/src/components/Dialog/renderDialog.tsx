import * as React from 'react';
import { DialogContext } from '../../dialogContext';
import type { DialogState } from './Dialog.types';

export const renderDialog = (state: DialogState) => {
  const { setOpen, toggleOpen, triggerRef, contentRef, trapFocus } = state;

  return (
    <DialogContext.Provider
      value={{
        setOpen,
        toggleOpen,
        triggerRef,
        contentRef,
        trapFocus,
      }}
    >
      {state.dialogTrigger}
      {state.open && state.dialogSurface}
    </DialogContext.Provider>
  );
};
