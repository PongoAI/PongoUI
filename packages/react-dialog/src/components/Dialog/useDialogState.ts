import * as React from 'react';
import { useControllableState, useEventCallback, useMergedRefs } from '@fluentui/react-utilities';
import type { DialogState, OpenDialogEvents } from './Dialog.types';

export const useDialogState = (state: DialogState) => {
  const { open, defaultOpen, onOpenChange } = state;
  const {
    onMouseEnter: onMouseEnterOriginal,
    onMouseLeave: onMouseLeaveOriginal,
    onKeyDown: onKeyDownOriginal,
  } = state.root;

  const dialogBoxRef = React.useRef(null);
  const contentRef = useMergedRefs(dialogBoxRef, state?.dialogBox?.ref);

  const [currentValue, setCurrentValue] = useControllableState({
    defaultState: open,
    state: defaultOpen,
    initialState: false,
  });

  const onDialogOpenChange = useEventCallback((ev: OpenDialogEvents, value: boolean) => {
    onOpenChange?.(ev, { open: value });
    setCurrentValue(!currentValue);
  });

  state.root.onMouseEnter = (e: React.MouseEvent<HTMLDivElement>) => {
    console.log('test');
    onMouseEnterOriginal?.(e);
  };

  state.root.onPointerDown = (ev: React.MouseEvent<HTMLDivElement>) => {
    console.log('test');
  };

  state.root.onMouseLeave = (e: React.MouseEvent<HTMLDivElement>) => {
    onMouseLeaveOriginal?.(e);
  };

  state.root.onKeyDown = (ev: React.KeyboardEvent<HTMLDivElement>) => {
    console.log('key down');

    if (ev.key === 'Escape' && contentRef.current?.contains(ev.target as HTMLDivElement)) {
      onDialogOpenChange(ev, false);
    }

    onKeyDownOriginal?.(ev);
  };

  return state;
};
