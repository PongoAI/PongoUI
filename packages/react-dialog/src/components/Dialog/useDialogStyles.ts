import { shorthands, makeStyles, mergeClasses } from '@griffel/react';
import { tokens } from '@pongo-ui/react-theme';
import type { DialogState } from './Dialog.types';

const useStyles = makeStyles({
  root: {
    position: 'fixed',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    inset: '0px',
    '-webkit-tap-highlight-color': 'transparent',
    transitionProperty: 'opacity',
    transitionDuration: tokens.transitionDuration,
    transitionTimingFunction: tokens.transitionTimingFunction,
    ':before': {
      content: "''",
      position: 'fixed',
      display: 'flex',
      inset: '0px',
      backgroundColor: tokens.textColor,
      opacity: '.5',
      '-webkit-tap-highlight-color': 'transparent',
    },
  },

  open: {
    opacity: '1',
  },

  closed: {
    opacity: '0',
  },

  dialogBox: {
    cursor: 'auto',
    display: 'inline-flex',
    backgroundColor: tokens.canvasColor,
    ...shorthands.padding(tokens.focusedLayout),
    flexDirection: 'column',
    flexShrink: 0,
    ...shorthands.gap(tokens.focusedLayout),
    filter: tokens.elevate,
    ...shorthands.borderRadius(tokens.rounded),
    outlineStyle: 'none',
  },
});

export const useDialogStyles = (state: DialogState): DialogState => {
  const styles = useStyles();

  state.root.className = mergeClasses(styles.root, state.open ? styles.open : styles.closed, state.root.className);

  if (state.dialogBox) {
    state.dialogBox.className = mergeClasses(styles.dialogBox, state.dialogBox.className);
  }

  return state;
};
