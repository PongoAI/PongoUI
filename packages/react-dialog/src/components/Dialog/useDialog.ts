import * as React from 'react';
import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
import { useModalAttributes } from '@fluentui/react-tabster';
import type { DialogProps, DialogState } from './Dialog.types';
import { useDialogState } from './useDialogState';

export const useDialog = (props: DialogProps, ref: React.Ref<HTMLDivElement>): DialogState => {
  const { open = true, dialogBox, onOpenChange, label } = props;

  const { modalAttributes } = useModalAttributes({ trapFocus: true });

  const state: DialogState = {
    open,
    label,
    onOpenChange,
    components: {
      root: 'div',
      dialogBox: 'div',
    },
    root: getNativeElementProps('div', {
      ref,
      role: 'dialog',
      'aria-modal': true,
      'aria-label': 'Hello world',
      ...modalAttributes,
      ...props,
    }),
    dialogBox: resolveShorthand(dialogBox, {
      required: true,
    }),
  };

  useDialogState(state);

  return state;
};
