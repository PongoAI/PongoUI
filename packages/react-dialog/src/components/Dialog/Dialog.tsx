import * as React from 'react';
import { useDialog } from './useDialog';
import { renderDialog } from './renderDialog';
import type { DialogProps } from './Dialog.types';

/**
 * The Dialog component helps manage state for the DialogSurface and DialogTrigger components.
 */
export const Dialog: React.FC<DialogProps> = props => {
  const state = useDialog(props);

  return renderDialog(state);
};
Dialog.displayName = 'Dialog';
