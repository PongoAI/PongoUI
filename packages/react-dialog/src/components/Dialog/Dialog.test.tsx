import React from 'react';
import { Dialog } from './Dialog';
import { DialogProps } from './Dialog.types';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('Dialog', () => {
  isConformant({
    Component: Dialog as React.FunctionComponent<DialogProps>,
    displayName: 'Dialog',
    disabledTests: ['component-has-root-ref', 'make-styles-overrides-win', 'component-handles-classname'],
    requiredProps: { open: true, label: 'This is a label' },
  });

  it('renders a basic Dialog', () => {
    const renderedComponent = render(
      <div>
        Test
        <Dialog open label="This is a label">
          Dialog content
        </Dialog>
      </div>,
    );

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });

  it('renders a closed Dialog', () => {
    const renderedComponent = render(
      <div>
        Test
        <Dialog open={false} label="This is a label">
          Dialog content
        </Dialog>
      </div>,
    );

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
