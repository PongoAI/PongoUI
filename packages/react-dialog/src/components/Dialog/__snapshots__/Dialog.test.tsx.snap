// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders a basic Dialog 1`] = `
<body>
  <div>
    <div>
      Test
      <span
        hidden=""
      />
    </div>
  </div>
  <div
    class=""
    dir="ltr"
  >
    <div
      aria-label="This is a label"
      aria-modal="true"
      class="___118frzz f19dog8a f22iagw f122n59 f4d9j23 f11qr6pd fmk5vbo f13u1uyl fkfijr0 fsxfru3 f99sooz f1k2v3fw fkmkm9e fu8n1pp fst2pvl f18r32vf f1g93lot f5p0z4x"
      data-tabster="{\\"deloser\\":{},\\"modalizer\\":{\\"id\\":\\"modal-3\\",\\"isOthersAccessible\\":false}}"
      role="dialog"
    >
      <div
        class="___zl8a6a0 f1ohdurq ftuwxu6 f8qjfyk fk7qxxv f10sqwl7 f1eg49sl fne32i3 f1vx9l62 fi64zpg fnmxc1e fynvm4a fl4n2cu f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1s6fcnf"
        tabindex="0"
      >
        Dialog content
      </div>
    </div>
  </div>
</body>
`;

exports[`Dialog renders a closed Dialog 1`] = `
<body>
  <div>
    <div>
      Test
      <span
        hidden=""
      />
    </div>
  </div>
  <div
    class=""
    dir="ltr"
  />
</body>
`;
